app/controllers/runtime/spaces_controller.rb

```diff
@@ -199,45 +199,29 @@ def delete(guid)
     delete '/v2/spaces/:guid/isolation_segment', :delete_isolation_segment
     def delete_isolation_segment(guid)
       space = find_guid(guid)
-      check_isolation_segment_access!(space)
-
-      check_space_is_empty!(space, 'Removing')
+      check_org_update_access!(space)
 
       space.db.transaction do
         space.lock!
-        space.update(isolation_segment_guid: nil)
-        space.save
+        space.update(isolation_segment_model: nil)
       end
 
       [HTTP::OK, object_renderer.render_json(self.class, space, @opts)]
     end
 
-    def before_update(obj)
+    def before_update(space)
       if request_attrs['isolation_segment_guid']
-        check_isolation_segment_access!(obj)
-
-        check_space_is_empty!(obj, 'Adding')
+        check_org_update_access!(space)
 
-        isolation_segment_guids = obj.organization.isolation_segment_models.map(&:guid)
-        unless isolation_segment_guids.include?(request_attrs['isolation_segment_guid'])
-          raise CloudController::Errors::ApiError.new_from_details('UnableToPerform',
-                                                                   'Adding the Isolation Segment to the Space',
-                                                                   "Only Isolation Segments in the Organization's allowed list can be used.")
-        end
+        raise CloudController::Errors::ApiError.new_from_details('ResourceNotFound', 'Isolation Segment not found') if
```

Comment:
This is a little confusing to read. It may be the go developer talking but I don't really like the if at the end of the line with no conditions. If you were to merge the condition in the same line it'd be a really long line. So I'd prefer if this was just a regular if around the `raise`.